promotion/src/main/java/org/cryptimeleon/incentive/promotion/PromotionDescription.java

```diff
@@ -0,0 +1,54 @@
+package org.cryptimeleon.incentive.promotion;
+
+
+import lombok.Getter;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Abstract class that represents a definition of a promotion.
+ */
+@Getter
+abstract public class PromotionDescription {
+    long promotionId;
+    String promotionTitle;
+    String promotionDescription;
+    LocalDate promotionStart;
+    LocalDate promotionEnd;
+    List<PromotionReward> promotionRewards;
```

Comment:
I don't think all instances of this need to have a finite list of possible rewards a priori. Some may just generate the possible rewards when seeing the basket content or token content (e.g., one reward could be to "cash out", for which there exist a lot of possible rewards, one for each possible point count).

Reply:
Okay, I would suggest replacing this by an abstract function that handles this in a more general way.

Reply:
Deferred to other branch, too :)